src/vs/editor/browser/controller/mouseTarget.ts

```diff
@@ -836,7 +836,10 @@ export class MouseTargetFactory {
 
 	private static _actualDoHitTestWithCaretRangeFromPoint(ctx: HitTestContext, coords: ClientCoordinates): IHitTestResult {
 
-		const range: Range = document.caretRangeFromPoint(coords.clientX, coords.clientY);
+		const docOrRoot = (window as any).monacoShadowRoot || document;
```

Comment:
There can be multiple editors instantiated in a single page, so using a global here is not possible.

Reply:
I have reworked this such that the dom node of the editor is checked to be in the shadow DOM or not dynamically.